providers/configure.rb

```diff
@@ -19,10 +19,19 @@
 #
 
 action :run do
+  sysconfig
   configure
   new_resource.updated_by_last_action(true)
 end
 
+def sysconfig
+  hugepage_file = "/sys/kernel/mm/transparent_hugepage/enabled"
+  file hugepage_file do
+    content 'never'
+    only_if { ::File.exists?(hugepage_file) && File.read(hugepage_file).chomp == 'never' }
```

Comment:
On a new Ubuntu Xenial VM, the output of this read command looks like
```
irb(main):001:0> File.read('/sys/kernel/mm/transparent_hugepage/enabled').chomp
=> "always [madvise] never"
```

It feels like there should be some kind of string parsing to determine which THP setting is active
```
irb(main):016:0> File.read('/sys/kernel/mm/transparent_hugepage/enabled').match(/^.*\[(.*)\]/).captures.first
=> "madvise"
```